server/models/authentication/auth.js

```diff
@@ -13,35 +12,44 @@ mongoose.connect(bocAuth, bocAuthOptions);
 
 const UserSchema = new mongoose.Schema({
   email: { type: String, required: true, unique: true },
-  password: { type: String, required: true },
-  firstName: { type: String, required: true},
-  lastName: { type: String, required: true}
+  password: { type: String, required: false },
+  firstName: { type: String, required: false},
+  googleId: { type: String, required: false},
+  lastName: { type: String, required: false}
 });
 
-var User = mongoose.model('User', UserSchema);
-
-var findUserByEmail = function (email) {
-  return User.findOne( {email: email} );
-}
-
-var addNewUser = function (data) {
-  findUserByEmail(data.email).then((user) => {
-    if (!user) {
-      var newUser = new User(data);
-      newUser.save(function (err) {
-        if (err) {
-          console.log(err);
-        };
-      });
-      console.log('new user created');
+var User = new mongoose.model('User', UserSchema);
+
+var findUserByEmail = async function (email, cb) {
+    try{
+      User.findOne( {email: email}, function (err, data) {
+        if (err){
+            console.log('findUserByEmail err:',err)
+            cb(err)
+        }
+        else{
+            cb(null, data)
+        }
+    });
+    } catch(err) {
+      cb(err);
```

Comment:
interesting, using error first callback pattern. Almost forgot about this one! ..assuming this worked better than a promise ..? It's clean and readable like this though.